src/main/java/com/cloudbees/jenkins/support/AsyncResultCache.java

```diff
@@ -37,29 +37,35 @@ public static <V, T extends java.lang.Throwable> V get(Node node, WeakHashMap<No
     public static <V, T extends java.lang.Throwable> V get(Node node, WeakHashMap<Node, V> cache, /*MasterToSlave*/Callable<V,T> operation, String name)
 
             throws IOException {
+        
         if (node == null) return null;
-        VirtualChannel channel = node.getChannel();
-        if (channel == null) {
-            synchronized (cache) {
-                return cache.get(node);
+        Future<V> future;
+        // If running in on the built-in node, no need to use the CallAsyncWrapper or be subjected 
+        // to the REMOTE_OPERATION_TIMEOUT_MS
+        if (node instanceof Jenkins) {
```

Comment:
Might be safer to call `JenkinsJVM.isJenkinsJVM()` if this might run on an agent? Not sure if `node instanceof Jenkins` will trigger classloading of the main `Jenkins` class on agents (which would be undesirable).

Reply:
The AsyncResultCache actually always executes on the controller JVM. It accepts the node and the callable and handle the asynchronous execution. So here we check if the node in which **we are going to** execute the callable is the controller or not.
Maybe I can rephrase my comment.